service/state/rpc.go

```diff
@@ -0,0 +1,109 @@
+package state
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/gorilla/mux"
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/celestiaorg/celestia-node/node/rpc"
+)
+
+var log = logging.Logger("state/rpc")
+
+var (
+	addrKey = "address"
+	txKey   = "tx"
+)
+
+func (s *Service) RegisterEndpoints(rpc *rpc.Server) {
+	rpc.RegisterHandlerFunc("/balance", s.handleBalanceRequest, http.MethodGet)
+	rpc.RegisterHandlerFunc(fmt.Sprintf("/balance/{%s}", addrKey), s.handleBalanceForAddrRequest, http.MethodPost)
+	rpc.RegisterHandlerFunc(fmt.Sprintf("/submit_tx/{%s}", txKey), s.handleSubmitTx, http.MethodPost)
+}
```

Comment:
These endpoints should be documented.

Reply:
https://github.com/celestiaorg/celestia-node/pull/591/commits/94d843d8902e64e90a55e9ef679fed035a0b0f93